src/Common/src/Interop/Unix/System.Native/Interop.MAdvise.cs

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime.InteropServices;
+
+using size_t = System.UIntPtr;
+
+internal static partial class Interop
+{
+    internal static partial class Sys
+    {
+        [DllImport(Libraries.SystemNative, SetLastError = true)]
+        internal static extern int MAdvise(IntPtr addr, size_t length, MemoryAdvice advice);
```

Comment:
I agree with (c)...I've been converting size_t (and other types like it) to long types; it seems much cleaner to me and makes an easy contract.